Sources/Sentry/SentryUIViewControllerSwizzling.m

```diff
@@ -261,10 +286,15 @@ - (void)swizzleRootViewControllerAndDescendant:(UIViewController *)rootViewContr
     for (UIViewController *viewController in allViewControllers) {
         Class viewControllerClass = [viewController class];
         if (viewControllerClass != nil) {
-            NSString *message = @"UIViewControllerSwizziling Calling swizzleRootViewController.";
-            [SentryLog logWithMessage:message andLevel:kSentryLevelDebug];
-
+            [SentryLog
+                logWithMessage:@"UIViewControllerSwizziling Calling swizzleRootViewController."
+                      andLevel:kSentryLevelDebug];
             [self swizzleViewControllerSubClass:viewControllerClass];
+
+            // We can't get the image name with the app delegate class for some apps. Therefore, we
+            // use the rootViewController and its subclasses as a fallback.  The following method
+            // ensures we don't swizzle ViewControllers of UIKit.
+            [self swizzleUIViewControllersOfClassesInImageOf:viewControllerClass];
```

Comment:
m: I know swizzling has duplication protection and usually the root view controller doesn't have many childViewControllers, but, can we improve this somehow? Let's say we have 4 child controllers, so we list all the binary controllers 4 times, filter them and try to swizzle.

Reply:
What about keeping track of the already swizzled images? We could skip then.

Reply:
Yes, keeping track of swizzled images would work.